### PR TITLE
[minor_change] Add support vrf ref as apic dn for mso_schema_site_bd_…

### LINKED ISSUE #497: Association between BD and L3out residing on APIC failling
### Description
* cannot associate a L3OUT on APIC (not imported into NDO) with BD stretched or local.

* Note: Association with a L3out on NDO (L3out container) is working if the schema and template are indicated under the l3out statement.

### Affected Module Name(s):
* mso_schema_site_bd_l3out

### MSO version and MSO Platform
* ND: 3.0.1i
* NDO: 4.2.3k

### APIC version and APIC Platform for Site Level Resources
* APIC: 5.3.2c

### Collection versions
* cisco.mso 2.8

### Output/ Error message
FAILED! => {
    "changed": true,
    "current": {
        "l3outName": "L3OUT_LOCAL_SITE1",
        "schemaId": "66196800196e988f095dc343",
        "templateName": "TMP_ALLSITES"
    },
    "data": [
        {
            "op": "replace",
            "path": "/sites/61469febeea713e579d584bf-TMP_ALLSITES/bds/BD1001",
            "value": {
                "bdRef": {
                    "bdName": "BD1001",
                    "schemaId": "66196800196e988f095dc343",
                    "templateName": "TMP_ALLSITES"
                },
                "hostBasedRouting": false,
                "l3OutRefs": [
                    "/schemas/66196800196e988f095dc343/templates/TMP_SITE1/l3outs/L3OUT_SITE1",
                    "/schemas/66196800196e988f095dc343/templates/TMP_ALLSITES/l3outs/L3OUT_LOCAL_SITE1"
                ],
                "l3Outs": [
                    "L3OUT_SITE1",
                    "L3OUT_LOCAL_SITE1"
                ],
                "mac": "00:22:BD:F8:19:FF",
                "subnets": []
            }
        }
    ],
    "info": {
        "body": {
            "code": 400,
            "info": null,
            "message": "Err Missing Ref: /schemas/66196800196e988f095dc343/templates/TMP_ALLSITES/l3outs/L3OUT_LOCAL_SITE1"
        },
        "connection": "close",
        "content-language": "en",
        "content-length": "135",
        "content-type": "application/json",
        "error": {
            "code": -1,
            "message": {
                "code": 400,
                "info": null,
                "message": "Err Missing Ref: /schemas/66196800196e988f095dc343/templates/TMP_ALLSITES/l3outs/L3OUT_LOCAL_SITE1"
            }
        },
        "method": "PATCH",
        "msg": "OK (135 bytes)",
        "server": "nginx",
        "status": 400,
        "strict-transport-security": "max-age=31536000; includeSubDomains",
        "url": "https://10.66.51.212:443/mso/api/v1/schemas/66196800196e988f095dc343?validate=false",
        "x-ratelimit-limit": "-1",
        "x-ratelimit-remaining": "-1",
        "x-ratelimit-reset": "1559582945"
    },
    "invocation": {
        "module_args": {
            "bd": "BD1001",
            "host": "10.66.51.212",
            "l3out": {
                "name": "L3OUT_LOCAL_SITE1",
                "schema": null,
                "template": null
            },
            "login_domain": null,
            "output_level": "normal",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": null,
            "schema": "HSM_TEMP_VRF01",
            "site": "CX-APJC-LAB-SITE1",
            "state": "present",
            "template": "TMP_ALLSITES",
            "timeout": null,
            "use_proxy": null,
            "use_ssl": null,
            "username": "user8-global",
            "validate_certs": false
        }
    },
    "msg": "MSO Error 400: Err Missing Ref: /schemas/66196800196e988f095dc343/templates/TMP_ALLSITES/l3outs/L3OUT_LOCAL_SITE1",
    "payload": {
        "code": 400,
        "info": null,
        "message": "Err Missing Ref: /schemas/66196800196e988f095dc343/templates/TMP_ALLSITES/l3outs/L3OUT_LOCAL_SITE1"
    },
    "status": 400
}

### Expected Behavior
* L3out associated with BD

### Actual Behavior
* error at playbook execution

### Playbook tasks to Reproduce
- name: Add L3out association on Bridge domain
  hosts: mso
  connection: local
  gather_facts: false

  tasks:
    - name: Add a new site BD l3out on site 1
      cisco.mso.mso_schema_site_bd_l3out:
        host: "{{ ansible_host }}"
        username: "{{ username }}"
        password: "{{ password }}"
        state: "present"
        validate_certs: False
        #use_ssl: True
        #use_proxy: True
        schema: "HSM_TEMP_VRF01"
        site: CX-APJC-LAB-SITE1
        template: TMP_ALLSITES
        bd: BD1001
        l3out:
          name: L3OUT_LOCAL_SITE1
        state: present

    - name: Add a new site BD l3out on site 2
      cisco.mso.mso_schema_site_bd_l3out:
        host: "{{ ansible_host }}"
        username: "{{ username }}"
        password: "{{ password }}"
        state: "present"
        validate_certs: False
        #use_ssl: True
        #use_proxy: True
        schema: "HSM_TEMP_VRF01"
        template: TMP_ALLSITES
        site: CX-APJC-LAB-SITE2
        bd: BD1001
        l3out:
          name: L3OUT_LOCAL_SITE2
        state: present

### References

* #0000

### PR DESCRIPTION
**Pull request by akinross**
…l3out

fixes #497

### CODE REVIEW

**Review by sajagana — DISMISSED** [review]
LGTM!

**Review by samiib — CHANGES_REQUESTED** [review]

File: plugins/modules/mso_schema_site_bd_l3out.py [hunk 1]
```diff
@@ -198,9 +206,17 @@ def main():
     payload = dict()
 
     if l3out:
-        l3out_schema_id = mso.lookup_schema(l3out.get("schema")) if l3out.get("schema") else mso_schema.id
-        l3out_template = l3out.get("template") if l3out.get("template") else template
-        l3out_ref = mso.l3out_ref(schema_id=l3out_schema_id, template=l3out_template, l3out=l3out.get("name"))
+
+        if l3out.get("tenant") and (l3out.get("schema") or l3out.get("template")):
```
[the remark this answers is not in the document]
Comment (akinross) [reply plugins/modules/mso_schema_site_bd_l3out.py:210]: How would this work when it is inside the l3out dict?

[the remark this answers is not in the document]
Comment (samiib) [reply plugins/modules/mso_schema_site_bd_l3out.py:210]: Hmm true I thought there would be a reason. 
However, based on the documentation (although not super clear) it appears to be possible.
Could `mutually_exclusive` be added to the l3out dictionary spec on line 176?

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by samiib — APPROVED** [review]
LGTM!

**Review by gmicol — APPROVED** [review]
LGTM

**Review by anvitha-jain — APPROVED** [review]

**Review by shrsr — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**